Sming/Arch/Esp8266/Core/HardwarePWM.cpp

```diff
@@ -23,115 +23,95 @@
  *
  */
 
+#include <HardwarePWM.h>
 #include <Clock.h>
 #include "ESP8266EX.h"
 #include <debug_progmem.h>
 
-#include <HardwarePWM.h>
-
 #define PERIOD_TO_MAX_DUTY(x) (x * 25)
 
 HardwarePWM::HardwarePWM(uint8_t* pins, uint8_t no_of_pins) : channel_count(no_of_pins)
```

Comment:
Please, keep close to Sming's coding style: https://sming.readthedocs.io/en/latest/information/develop/coding-style.html?highlight=Variables#naming. 
```
Local variables, instance variables, and class variables must also be written in lowerCamelCase. Variable names must not start with, end with or contain underscore (_) or dollar sign ($) characters. 
```

For example in the code here `uint8_t no_of_pins` should be renamed to `uint8_t numberOfPins`, `uint32_t io_info[PWM_CHANNEL_NUM_MAX][3];` -> `uint32_t ioInfo[PWM_CHANNEL_NUM_MAX][3];`,  `uint32_t pwm_duty_init[PWM_CHANNEL_NUM_MAX]` -> `uint32_t pwmDutyInit[PWM_CHANNEL_NUM_MAX]` and so on.

Reply:
> should this be part of this PR or may be a PR of it's own

Leave it for another PR. This one waited already for too long to be merged. 

> I believe, all other occurences of underscored names are defined in the esp32 ledc interface.

Please, rename all local variables in the code that you have introduced to use lowerCamelCase . Example:

```C++
	uint32_t io_info[PWM_CHANNEL_NUM_MAX][3];	// pin information
	uint32_t pwm_duty_init[PWM_CHANNEL_NUM_MAX]; // pwm duty
  // ...
	const int initial_period = 1000;

```

and so on.

Reply:
I don't use those, they are part of the original ESP8266 version and are used both there and in the RP2040 implementation. I'll go ahead and rename them all and send a separate PR.

Reply:
so, after looking at this a bit more, I believe what has happened.
The only files I have changed for this should be Arch/Esp32/Core/HardwarePWM.cpp and Core/HardwarePWM.h 
At some point, I did run make cs which has touched a ton of files throughout the whole source tree. I believe that's how the Arch/Esp8266 files got into this pr. They should not be there. 
At the risk that I'm now being more work than it was to write the original pwm implementation, I guess it would be best to just rebase the changes I did and leave the rest as it is in the upstream develop branch. I'm most happy to implement the changes suggested there, but they really should not be in this PR.

As a side note: I am very impressed with the level of professionality and support you are applying here, and I really appreciate your help a lot, even though it makes me feel very much like a total noob.